cypress/pageobjects/ClusterMachinePoolDetails.page.js

```diff
@@ -174,12 +188,81 @@ class ClusterMachinePoolDetails extends Page {
   }
 
   isOverviewClusterPropertyMatchesValue(property, value) {
-    cy.get('span.pf-v6-c-description-list__text')
-      .contains(property)
-      .parent()
-      .siblings()
-      .find('div')
-      .contains(value);
+    cy.get('span').contains(property).parent().siblings().find('div').contains(value);
+  }
+
+  // Machine Pool Modal Verification Methods
+  verifyMachinePoolModalIsOpen() {
+    // Verify the modal is visible and accessible
+    cy.get('#edit-mp-modal').should('be.visible');
+
+    return this;
+  }
+
+  verifyMachinePoolModalTitle(expectedTitle) {
+    // Verify modal title matches expected text
+    cy.get('h1').contains(expectedTitle);
+  }
+
+  verifySubTabExists(tabName) {
+    // Use regex for flexible matching, especially for "Labels" in "Labels and Taints"
+    if (tabName.toLowerCase().includes('labels')) {
+      const labelsRegex = /^Labels\b/i;
+      cy.get('[role="tab"]').contains(labelsRegex).should('be.visible');
+    } else {
+      cy.get('[role="tab"]').contains(tabName).should('be.visible');
+    }
+    return this;
+  }
+
+  selectMachinePoolSubTab(tabName) {
+    // Use regex for flexible matching for Labels tab
+    if (tabName.toLowerCase().includes('labels')) {
+      const labelsRegex = /^Labels\b/i;
+      cy.get('[role="tab"]').contains(labelsRegex).click();
+    } else {
+      cy.get('[role="tab"]').contains(tabName).click({ force: true });
```

Comment:
Could we keep more text oriented locator definition in this case? For example, `cy.get('button').contains(tabName)`